src/cmdlib.sh

```diff
@@ -487,28 +484,25 @@ EOF
 
     local runvm_console
     runvm_console="${workdir}/tmp/runvm-console.txt"
-    rm -f "${workdir}/tmp/rc" "${runvm_console}" "${cmdoutput}"
+    rm -f "${workdir}/tmp/rc" "${runvm_console}"
 
     touch "${runvm_console}"
-    setpriv --pdeathsig SIGTERM -- tail -qF "${cmdoutput}" --pid $$ &
     #shellcheck disable=SC2086
     kola qemuexec -m 2048 --auto-cpus -U --workdir none -- -no-reboot -nodefaults -serial file:"${runvm_console}" \
         -kernel "${vmbuilddir}/kernel" -initrd "${vmbuilddir}/initrd" \
+        -device virtio-serial \
+        -device virtserialport,chardev=virtioserial0,name=cosa-cmdout \
+        -chardev stdio,id=virtioserial0 \
         -drive "if=virtio,format=raw,snapshot=on,file=${vmbuilddir}/root,index=1" \
         "${cachedisk[@]}" \
         -virtfs local,id=workdir,path="${workdir}",security_model=none,mount_tag=workdir \
         "${srcvirtfs[@]}" -append "root=/dev/vda console=${DEFAULT_TERMINAL} selinux=1 enforcing=0 autorelabel=1" \
-	"${qemu_args[@]}"
-    kill %1
+        "${qemu_args[@]}" <&- || true
```

Comment:
I had to look up `<&-` - is that "standard"?  Feels hostile to a process to have fd `0` actually be closed instead of e.g. `</dev/null`.

Reply:
Also I think it'd be cleaner to do:

```
if ! kola qemuexec ... ; then
  cat "${runvm_console}"
  fatal "qemu failed"
fi
```

so the case of "qemu died" is more clearly distinguished from "init died".

Reply:
> I had to look up <&- - is that "standard"?

Depends what you mean by "standard". :) It's in the bash manual at least: https://www.tldp.org/LDP/abs/html/io-redirection.html, but I'm not sure if it's POSIX shell (though we already use bash-isms a bit everywhere in this codebase).

> Feels hostile to a process to have fd 0 actually be closed instead of e.g. </dev/null.

Oddly, `</dev/null` actually triggers the issue for me locally and I'm not sure why. Closing stdin works locally and in CI.

> Also I think it'd be cleaner to do:

Yup sure, done! (The `|| true` bit here was actually leftover from debugging, I didn't intend to commit that. But your suggestion makes sense regardless.)

Reply:
> Oddly, </dev/null actually triggers the issue for me locally and I'm not sure why. Closing stdin works locally and in CI.

Weird, OK well we can improve this later.